### ISSUE TITLE
What would be the process of finetuning Jasper on domain specific data?

### ISSUE THREAD

**Issue by TheBrownViking20** [NVIDIA/OpenSeq2Seq#470]
I have some financial data and I want to use transfer learning to fine tune it for financial data speech to text capability. Is some method available as a part of the toolkit? If not, then how do I go about doing this?

**Comment by blisc** [comment]
#452 has some starting points on fine tuning.

I would be curious if LibriSpeech-trained Jasper would work if you simply create a n-gram language model off your financial data and try to decode with LM.

**Comment by borisgin** [comment]
- convert audio to "mono"
-  if you want to use n-gran language model use KenLM 
- put checkpoint into log directtory
- convert data to the format used for training, see  "scripts/import_librivox.py" 
- modify Jasper.config [Jasper.config](https://github.com/NVIDIA/OpenSeq2Seq/blob/master/example_configs/speech2text/jasper10x5_LibriSpeech_nvgrad.py):
train_params = {
    "data_layer": Speech2TextDataLayer,
    "data_layer_params": {
        "dataset_files": [
          **"your_data.csv"**
         ]
}
- set small learning rate
-train with  "-continue_learning" flag

**Comment by tayciryahmed** [comment]
@borisgin  are you sure the learning rate is re-initialized with `continue_learning` ?

**Comment by ngochuyenluu** [comment]
@blisc: Hi, I used the pretrained of Librispeech on Jasper to train on my own data which is about finance and tax. The results are very limited, and I am trying to improve the results with N-gram language model, have you tested it before because I have questions about which files to configure in command:
https://github.com/NVIDIA/OpenSeq2Seq/tree/master/external_lm_rescore

python scripts/decode.py --logits=model_output.pickle --labels="CSV_FILE" --lm="LM_BINARY"  --vocab="ALPHABET_FILE" --alpha=ALPHA --beta=BETA --beam_width=BEAM_WIDTH --dump_all_beams_to=BEAMS.txt

lm, vocab are taken by toy_speech_data? and do you have any advice to tunne alpha, beta, beam_with?

**Comment by ngochuyenluu** [comment]
@bllsc and @borisgin : Hi, it's still me, I was curious about you approach also, that's why I continued from the pretrained-model Jasper on Librispeech, use n-gram language model and beam search on our tax and financial dataset with a small learning rate and increase of number of epochs... I have replaced my dataset in train_params, eval_params, infer_params by our training, dev and test files.

The model improved well on training set, but the validation set is poor. But I tested the pretrained model on the same validation set, it was better than the training I was doing.

My question is this possible to continue from pretrained model on Librispeech beacause it will gain a lot of efforts? (I have a impression is None), if yes, could you please gave us some details about how to train the dataset on pretrained to give the better results? Is there any problem when I changed the dataset and it made change of checkpoints that make the model getting poorly?

Thank you in advance for your answers, any recommendations will help us a lot.
